### PR TITLE
Adding prettier

### PR DESCRIPTION
**Pull request by sbeih**
- Added Prettier.
- Formatted all files.
- VSCode on-save prettify configs.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "trailingComma": "es5",
+    "tabWidth": 2,
```
Comment (ron-b) [inline .prettierrc:3]: Lets do tabWidth 4

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    "semi": true,
+    "singleQuote": true
```
Comment (ron-b) [inline .prettierrc:5]: I prefer keeping double quotes

File: package.json [hunk 3]
```diff
@@ -41,10 +42,12 @@
     "@typescript-eslint/parser": "^4.28.2",
     "babel-jest": "^27.0.6",
     "eslint": "^7.30.0",
+    "eslint-config-prettier": "^8.3.0",
     "eslint-plugin-jest": "^24.3.6",
+    "eslint-plugin-prettier": "^3.4.0",
     "jest": "^27.0.6",
     "nyc": "^15.1.0",
-    "prettier": "^2.2.1",
+    "prettier": "2.3.2",
```
Comment (ron-b) [inline package.json:50]: any reason not to use `^2.3.2`?

**Review by ron-b — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (sbeih) [reply package.json:50]: Making sure that everyone will get the same version, because even small patches in prettier can make a some difference.

**Review by ron-b — APPROVED** [review]

**Pull request merged**